### PR TITLE
[core] Move `src/ray/common/status.h` to `src/ray/util/status.h`

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?

Status only depends on util and no other things in common. Common is very complicated right now and it has many dependences.

For library doesn't depend on other things in common, it'll cost a lot of time to compile.

This PR moves the status to util since this should be the best place for it given the reason above.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
How do we decide which should put inside common and which inside util?

My general guideline is that things inside util folder are not project (e.g. Ray) specific, they can be used by other projects as well, a good example is `ray/util/ordered_set.h`

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -989,30 +988,33 @@ void NodeManager::NodeAdded(const GcsNodeInfo &node_info) {
...
-  // Update the resource view if a new message has been sent.
-  if (RayConfig::instance().use_ray_syncer()) {
-    if (auto sync_msg = ray_syncer_.GetSyncMessage(node_id.Binary(),
-                                                   syncer::MessageType::RESOURCE_VIEW)) {
-      if (sync_msg) {
-        ConsumeSyncMessage(sync_msg);
-      }
-    }
-  }
+  // Fetch resource info for the remote node and update cluster resource map.
```
Comment (rkooo567) [inline src/ray/raylet/node_manager.cc:991]: why is this file changed? Is it accidental?

**Comment by iycheng** [comment]
@jjyao @rkooo567  I think this is not the right way to do it. The issue here is that BUILD is not modularized. Let me close this one and fix it in the other way.
